### PR TITLE
Add has map packet for spectator

### PR DESCRIPTION
**Pull request by ascenttree**
This pr adds support for the "has map" packet in spectator, resolving #19.

### CODE REVIEW

File: hnet/constants.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ const (
 	CLIENT_REQUEST_STATS       uint32 = 11
 	CLIENT_START_SPECTATING    uint32 = 14
 	CLIENT_STOP_SPECTATING     uint32 = 15
+	CLIENT_HAS_MAP             uint32 = 16
```
Comment (Lekuruu) [inline hnet/constants.go:10]: I would probably name it something like `CLIENT_SPECTATOR_HAS_MAP` & `SERVER_SPECTATOR_HAS_MAP`

File: hnet/writers.go [hunk 2]
```diff
@@ -106,6 +106,17 @@ func (request SpectateRequest) Serialize(stream *common.IOStream) {
 	stream.WriteU32(request.UserId)
 }
 
+func (request HasMapRequest) Serialize(stream *common.IOStream) {
+	stream.WriteU8(1)
+	stream.WriteU32Bool(request.HasMap)
+}
+
+func (response HasMapResponse) Serialize(stream *common.IOStream) {
+	stream.WriteU32(2)
```
Comment (Lekuruu) [inline hnet/writers.go:115]: Why is this 2 specifically?

Comment (ascenttree) [reply hnet/writers.go:115]: Hexis reads the user id and boolean as a list, so there is 2 elements

File: hnet/constants.go (hunk 1 above)
Comment (ascenttree) [reply hnet/constants.go]: Done

File: hnet/writers.go (hunk 2 above)
Comment (Lekuruu) [reply hnet/writers.go:115]: Wow, this explains a *lot*...

File: hnet/handler.go [hunk 3]
```diff
@@ -183,6 +183,24 @@ func handleStopSpectating(stream *common.IOStream, player *Player) error {
...
+		return fmt.Errorf("failed to read has map request")
+	}
+
+	player.LogIncomingPacket(CLIENT_SPECTATE_HAS_MAP, request)
+
+	response := &HasMapResponse{
+		UserId: player.Info.Id,
+		HasMap: request.HasMap,
+	}
+	player.Host.SendPacket(SERVER_SPECTATE_HAS_MAP, response)
```
Comment (Lekuruu) [inline hnet/handler.go:199]: We also need to send this to the host's spectators. So `player.Host.Spectators.Broadcast`.

Comment (Lekuruu) [reply hnet/handler.go:199]: Nevermind, I forgot that there were no fellow spectators.. whoops

**Review by Lekuruu — APPROVED** [review]

**Pull request merged**